### PR TITLE
Pypi Release 0.1.0

### PR DESCRIPTION
**Pull request by rowanc1**

### CODE REVIEW

**Comment by rowanc1** [comment]
@lheagy can you take a quick look at this before I merge to master.

`pip install discretize` should work now.

You may need an updated pymatsolver. (also works on pip)

**Comment by lheagy** [comment]
It doesn't look like the pip install is working (microsoft azure notebooks - runs linux) 

image

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+clean:
+	cd docs;make clean
+	find . -name "*.pyc" | xargs -I {} rm -v "{}"
+
+clean_cython: clean
+	find . -name "*.c" | xargs -I {} rm -v "{}"
+	find . -name "*.so" | xargs -I {} rm -v "{}"
+
+deploy:
+	python setup.py sdist bdist_wheel upload
```
Comment (lheagy) [inline Makefile]: do we want to use `twine` ?
https://packaging.python.org/distributing/

image

File: discretize/utils/interputils.py [hunk 2]
```diff
@@ -66,99 +66,3 @@ def interpmat(locs, x, y=None, z=None):
     Q = sp.csr_matrix((vals, (I, J)),
                       shape=(npts, np.prod(shape)))
     return Q
-
-if not _interpCython:
```
Comment (lheagy) [inline discretize/utils/interputils.py]: Until we get the pip install stable with cython, I think we should hang on to this code...

File: docs/content/installing.rst [hunk 3]
```diff
@@ -19,6 +19,7 @@ Development Dependencies
 - sphinx
 - sphinx_rtd_theme
 - sphinx-gallery
```
Comment (lheagy) [inline docs/content/installing.rst]: should we pull these into a dev-requirements.txt (sphinx and sphinx-gallery are not strictly necessary for discretize)

File: discretize/utils/meshutils.py [hunk 4]
```diff
@@ -181,11 +181,11 @@ def ExtractCoreMesh(xyzlim, mesh, meshType='tensor'):
     This function ouputs::
 
         - actind: corresponding boolean index from global to core
-        - meshcore: core SimPEG mesh
+        - meshcore: core mesh
 
     Warning: 1D and 2D has not been tested
     """
-    from SimPEG import Mesh
+    from discretize import Mesh
```
Comment (lheagy) [inline discretize/utils/meshutils.py]: this will fail

File: docs/content/installing.rst (hunk 3 above)
Comment (rowanc1) [inline docs/content/installing.rst]: they are in a req_dev.txt already.

File: discretize/utils/meshutils.py (hunk 4 above)
Comment (rowanc1) [inline discretize/utils/meshutils.py]: fixing this. (it is not tested....)

File: docs/content/installing.rst (hunk 3 above)
Comment (lheagy) [inline docs/content/installing.rst]: kk, do we want them in the `installing.rst`?

**Comment by rowanc1** [comment]
@lheagy I got the build semi working on appveyor. Looks like the cython is installing at least.

There is a problem in the TreeMesh with integer overflows, otherwise things seem to work. I am not sure if the `pip` install will work on windows, but it does work on linux now.

I am going to leave the int overflow (I think it needs to be 64bit?) for another time.

File: .bumpversion.cfg [hunk 5]
```diff
@@ -1,4 +1,4 @@
 [bumpversion]
-current_version = 0.0.5
+current_version = 0.1.1b1
```
Comment (lheagy) [inline .bumpversion.cfg]: does this do the `--pre`?

Comment (rowanc1) [inline .bumpversion.cfg]: yeah

**Comment by rowanc1** [comment]
image

**Pull request merged**